### PR TITLE
committing rtsock healthbot rules.

### PR DESCRIPTION
**Pull request by anchal26**

### CODE REVIEW

**Comment by gvarao** [comment]
jenkins please retry a build

**Comment by prashant3167** [comment]
jenkins please retry a build

**Comment by gvarao** [comment]
jenkins please retry a build

File: juniper_official/Kernel/README.md [hunk 1]
```diff
@@ -383,3 +383,66 @@ jlock hogs or TCP connection drop messages get reported in these log files
 around the same time. Report to Juniper if this is happening consistently
 or if there is a noticeable side effect seen (such as protocol flaps / traffic
 loss) around the same time.
+
```
Comment (gvarao) [inline juniper_official/Kernel/README.md]: This file is conflicting, please resolve issue

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/*
+ * This rule checks for the total number of routing socket errors at any given time in the system.
+ * Routing socket errors includes errors returned from rtsock, veto and mproc layers.
+ * It monitors the total number of routing socket errors and raises an alarm if there are too many of
+ * these detected for significant duration of time.
+ *
+ * This rule will check if the error has benn increased by 100 from its previous value in ALL the samples
+ * for the last 70 or 130 seconds and raises the alarm accordingly.
+ */
+iceberg {
```
Comment (gvarao) [inline juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: Replace iceberg with healthbot

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    topic kernel.rtsock {
+        rule detect-routing-socket-errors-iagent {
+            synopsis "Detect Routing Socket errors in the system.";
```
Comment (gvarao) [inline juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: Remove periods from all descriptions and synopsis in all rules and playbooks

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            description "Detect errors in rtsock, veto and mproc layers.";
```
Comment (gvarao) [inline juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: Remove period

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            sensor rtsock-error-count-sensor {
+                synopsis "Rtsock error count sensor.";
+                description "sensor to keep track of rtsock error counts.";
```
Comment (gvarao) [inline juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: Start description with CAPS letter and remove period

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                iAgent {
+                    file rtsock-error-count.yml;
+                    table getRtsockErrorCountTable;
+                    frequency 20s;
```
Comment (gvarao) [inline juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: If the rule is not mission critical, please increase sensor frequency

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                }
+            }
+            field rtsock-error-count-threshold {
+                constant {
+                    value "{{rtsock-error-count-threshold-input}}";
+                }
```
Comment (gvarao) [inline juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: Field type is mandatory hence specify type.

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                sensor rtsock-error-count-sensor {
+                    path total_veto_count;
+                }
+                type integer;
+                description "Total rtsock veto count.";
+            }
+            trigger report-rtsock-errors-iagent {
+                synopsis "Trigger to check rtsock error count.";
+                description "Trigger to check rtsock error count";
+                frequency 20s;
```
Comment (gvarao) [inline juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: Ensure trigger frequency is minimum 1.5 times of sensor frequency in all rules

File: juniper_official/Kernel/detect-routing-socket-errors.rule [hunk 9]
```diff
@@ -0,0 +1,124 @@
+/*
+ * This rule checks for the total number of routing socket errors at any given time in the system.
+ * Routing socket errors includes errors returned from rtsock, veto and mproc layers.
+ * It monitors the total number of routing socket errors and raises an alarm if there are too many of 
+ * these detected for significant duration of time. 
+ * 
+ * This rule will check if the error has been increased by 100 from its previous value in ALL the samples 
+ * for the last 70 or 130 seconds and raises the alarm accordingly.
+ */
+iceberg {
```
Comment (gvarao) [inline juniper_official/Kernel/detect-routing-socket-errors.rule]: Please address all above comments i.e. related field type, removal of periods, iceberg to healthbot, sensor frequency, trigger frequency, etc. in all rules

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule (hunk 6 above)
Comment (anchal26) [reply juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: This is a critical parameter. Hence the frequency is 20s.

File: juniper_official/Kernel/detect-routing-socket-errors-iagent.rule (hunk 7 above)
Comment (anchal26) [reply juniper_official/Kernel/detect-routing-socket-errors-iagent.rule]: This is a constant field.

File: juniper_official/Kernel/README.md (hunk 1 above)
Comment (PrateekDilip) [reply juniper_official/Kernel/README.md]: This README.md was added from my commit. But in the latest my README is not present. You can treat this README as a new file. Talked to Anchal, dance resolving the comment.

**Comment by gvarao** [comment]
jenkins please retry a build

**Pull request merged**